school/src/main/java/com/hogwarts/school/controller/StudentController.java

```diff
@@ -17,38 +19,43 @@ public StudentController(StudentService studentService) {
         this.studentService = studentService;
     }
 
-    @PostMapping("/addStudent")
+    @PostMapping
     public ResponseEntity<Student> addStudent(@RequestBody Student student) {
-        Student std = studentService.addStudent(student);
-        return ResponseEntity.ok(std);
+        return ResponseEntity.ok(studentService.createStudent(student));
     }
 
-    @GetMapping("/getStudent/{studentId}")
+    @GetMapping("{studentId}")
     public ResponseEntity<Student> getStudent(@PathVariable Long studentId) {
-        Student std = studentService.getStudent(studentId);
-        if (studentId == null) {
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(std);
+        return ResponseEntity.ok(studentService.findStudent(studentId));
     }
 
-    @PutMapping()
-    public ResponseEntity<Student> updateStudent(@RequestBody Student student) {
-        Student std = studentService.updateStudent(student);
-        if (std == null) {
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(std);
+    @PutMapping("/{id}")
+    public ResponseEntity<Student> updateStudent(@PathVariable Long id, @RequestBody Student student) {
+        return ResponseEntity.ok(studentService.updateStudent(id,student));
     }
 
-    @DeleteMapping("/removeStudent/{studentId}")
-    public ResponseEntity<Void> removeStudent(@PathVariable Long studentId) {
-        studentService.removeStudent(studentId);
-        return ResponseEntity.ok().build();
+    @DeleteMapping("/{studentId}")
+    public ResponseEntity<Student> removeStudent(@PathVariable Long studentId) {
+        return ResponseEntity.ok(studentService.removeStudent(studentId));
     }
-    @GetMapping("getStudentByAge/{studentId}")
-    public ResponseEntity<List<Student>> getStudentByAge(@PathVariable int studentAge) {
-        List<Student> std = studentService.filteringStudentsByAge(studentAge);
+    @GetMapping(params = "studentAge")
+    public ResponseEntity<Collection<Student>> getStudentByAge(@PathVariable int studentAge) {
```

Comment:
```suggestion
    public ResponseEntity<Collection<Student>> getStudentByAge(@RequestParam int studentAge) {
```
здесь не будет работать, аннотация не та

Reply:
по-прежнему актуально